aws/cloud-init.tf

```diff
@@ -0,0 +1,137 @@
+//
+// Cloud-init scripts
+//
+
+// Common cloud-init configuration for all nodes
+data "template_file" "common-config" {
+  template = "${file("${path.module}/cloud-init/common.config.tpl")}"
+
+  vars {
+    cluster_name         = "${var.name}"
+    gravity_version      = "${var.gravity_version}"
+    source               = "${var.dl_url}"
+    ops_url              = "${var.ops_url}"
+    ops_token            = "${var.ops_token}"
+    flavor               = "${var.flavor}"
+    master_role          = "${var.master_role}"
+    worker_role          = "${var.worker_role}"
+    ops_advertise_addr   = "${var.ops_advertise_addr}"
+    aws_hosted_zone_name = "${var.aws_hosted_zone_name}"
+    email                = "${var.email}"
+    skip_install         = "${var.skip_install}"
+  }
+}
+
+// Common script file to be executed on all nodes
+data "template_file" "common-script" {
+  template = "${file("${path.module}/cloud-init/common.script.tpl")}"
+
+  vars {
+    cluster_name         = "${var.name}"
+    gravity_version      = "${var.gravity_version}"
+    source               = "${var.dl_url}"
+    ops_url              = "${var.ops_url}"
+    ops_token            = "${var.ops_token}"
+    flavor               = "${var.flavor}"
+    master_role          = "${var.master_role}"
+    worker_role          = "${var.worker_role}"
+    ops_advertise_addr   = "${var.ops_advertise_addr}"
+    aws_hosted_zone_name = "${var.aws_hosted_zone_name}"
+    email                = "${var.email}"
+    skip_install         = "${var.skip_install}"
+  }
+}
+
+// Shell script for bootstrapping master nodes
+data "template_file" "master" {
+  template = "${file("${path.module}/cloud-init/master.script.tpl")}"
+
+  vars {
+    cluster_name         = "${var.name}"
+    gravity_version      = "${var.gravity_version}"
+    source               = "${var.dl_url}"
+    ops_url              = "${var.ops_url}"
+    ops_token            = "${var.ops_token}"
+    flavor               = "${var.flavor}"
+    master_role          = "${var.master_role}"
+    worker_role          = "${var.worker_role}"
+    ops_advertise_addr   = "${var.ops_advertise_addr}"
+    aws_hosted_zone_name = "${var.aws_hosted_zone_name}"
+    email                = "${var.email}"
+    skip_install         = "${var.skip_install}"
+
+    // OIDC variables
+    oidc_client_id     = "${var.oidc_client_id}"
```

Comment:
instead of specific variables here I would prefer to use generic resource file that will get re-applied to telekube on every run

Reply:
As mentioned on slack, that's temporary... OIDC / gravity resource creation will move to our provider plugin when we get to it. For now though, it's a convenient way to use OIDC when testing.